source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+ * on a user-defined adiabatic boundary for the 3D ellipsoid
+ * chunk geometry model. It discretizes the model domain into
+ * two regions separated by an isotherm boundary below which
+ * the temperature increases adiabatically. Above the
+ * user-defined isotherm boundary the temperature linearly
+ * increases from a surface temperature (273.15 K or 0 degree C)
+ * to the isotherm (1673.15 K or 1600 degree C). The user
+ * defines the location of the isotherm boundary with an ascii
+ * data file with the format defined in the ASPECT manual. Note
+ * that the latitudinal and longitudinal bounds of the ascii input
+ * data file need to be at least 1 degree wider than the bounds
+ * you use to define the region of your ellipsoid chunk geometry.
+ *
+ * This plugin is developed by Tahiry Rajaonarison, D. Sarah Stamps,
+ * and Wolfgang Bangerth.
+ */
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    using namespace dealii;
+
+    /**
+     * Function that returns latitude and longitude from ECEF Cartesian
+     * coordinates that account for ellipsoidal shape of the Earth
+     * with WGS84 parameters.
+     */
+
+    template <int dim>
+    std::pair<double, double>
+    AdiabaticBoundary<dim>::lat_long_from_xyz_WGS84(const Point<3> &pos) const
+    {
+      /* Define WGS84 ellipsoid constants. */
+      const double radius = 6378137;
+      const double ellipticity = 8.1819190842622e-2;
+
+      const double b = std::sqrt(radius * radius
+                                 * (1 - ellipticity * ellipticity));
+      const double ep = std::sqrt((radius * radius - b * b) / (b * b));
+      const double p = std::sqrt(pos(0) * pos(0) + pos(1) * pos(1));
+      const double th = std::atan2(radius * pos(2), b * p);
+      const double lon = std::atan2(pos(1), pos(0));
+      const double lat = std::atan2((pos(2) + ep * ep * b * std::sin(th)
+                                     * std::sin(th) * std::sin(th)),
+                                    (p - (ellipticity
+                                          * ellipticity
+                                          * radius
+                                          * (std::cos(th) * std::cos(th)
+                                             * std::cos(th)))));
+
+      /* Convert to degrees. */
+      const double lon_degrees = lon * (180 / numbers::PI);
+      const double lat_degrees = lat * (180 / numbers::PI);
+
+      /* Set all longitudes between [0,360]. */
+      if (lon_degrees < 0)
+        return std::make_pair(lat_degrees, lon_degrees + 360);
+      else if (lon_degrees > 360)
+        return std::make_pair(lat_degrees, lon_degrees - 360);
+      else
+        return std::make_pair(lat_degrees, lon_degrees);
+
+    }
+
+    template <>
+    double
+    AdiabaticBoundary<2>::get_isotherm_depth(const double,
+                                             const double) const
+    {
+      AssertThrow(false, ExcInternalError());
```

Comment:
Please make the exception message more specific (like "The 'adiabatic boundary' initial temperature plugin is only implemented for 3d cases."). You can do that by using ExcMessage("Message") instead of ExcInternalError().